targets/targets.json

```diff
@@ -612,6 +612,17 @@
         "device_name": "MK64FN1M0xxx12",
         "bootloader_supported": true
     },
+    "EV_COG_AD4050LZ": {
+        "inherits": ["Target"],
+        "core": "Cortex-M4",
+        "supported_toolchains": ["ARM", "GCC_ARM", "IAR"],
+        "macros": ["__ADUCM4050__", "EV_COG_AD4050LZ"],
+        "extra_labels": ["Analog_Devices", "ADUCM4X50", "ADUCM4050", "EV_COG_AD4050LZ", "FLASH_CMSIS_ALGO"],
+        "device_has": ["SERIAL", "STDIO_MESSAGES", "TRNG", "SLEEP", "INTERRUPTIN", "LOWPOWERTIMER", "RTC", "SPI", "I2C", "FLASH", "ANALOGIN"],
```

Comment:
I2C is not fully implemented, why are some API missing? This might be case for some other HAL implementations (some functions are empty).

Reply:
The I2C API's that are not implemented are due to the I2C hardware handling features such as start and stop.  There is no way to create only a start or stop condition.  Functions such as these are usually used in software I2C implementations.

Reply:
This should be stated in the code either via comment or git commit msg, I can see lot of empty functions but do not know the reason behind those (to support return 0 for instance). It  would be helpful

Reply:
Comments added to I2C and SPI modules as recommended.